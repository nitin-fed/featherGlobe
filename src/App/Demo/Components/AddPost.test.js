import React from "react";
import renderer from "react-test-renderer";
import { AddPost } from "./AddPost";

describe("Add Posts", () => {
  const props = {
    isVisible: true,
    isEditing: true,
    cancelHandler: jest.fn(),
    submitHandler: jest.fn(),
    postId: 1,
    posts: [
      {
        body: "test post body",
        id: 1,
        title: "test post title",
        userId: 1
      },
      {
        body: "test post body",
        id: 2,
        title: "test post title",
        userId: 2
      }
    ]
  };
  test("Should render isEditing  === true", () => {
    const component = renderer.create(<AddPost {...props} />);
    let tree = component.toJSON();
    //expect(tree).toMatchSnapshot();
  });

  test("Should render if isEditing === false", () => {
    const newProps = {
      ...props,
      isEditing: false
    };
    const component = renderer.create(
      <AddPost isEditing="false" {...newProps} />
    );
    let tree = component.toJSON();
    //expect(tree).toMatchSnapshot();
  });

  test("Should render if isVisible === false", () => {
    const newProps = {
      ...props,
      isVisible: false
    };
    const component = renderer.create(
      <AddPost isEditing="false" {...newProps} />
    );
    let tree = component.toJSON();
    //expect(tree).toMatchSnapshot();
  });
});
