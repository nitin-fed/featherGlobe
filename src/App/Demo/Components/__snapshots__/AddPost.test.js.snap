// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Posts Should render if isEditing === false 1`] = `
<div
  className="addPostForm"
>
  <div>
    <div
      className="postTitle"
    >
      Add
       New Post
    </div>
    <form>
      <div>
        <label>
          ID:
        </label>
        <input
          disabled={true}
          type="text"
        />
        <br />
        <label>
          Title:
        </label>
         
        <input
          placeholder="Enter Title"
          type="text"
        />
         
        <br />
        <label>
          Description:
        </label>
        <textarea
          placeholder="Description"
          rows="5"
        />
      </div>
      <br />
      <button
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        onClick={[MockFunction]}
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;

exports[`Add Posts Should render if isVisible === false 1`] = `null`;

exports[`Add Posts Should render isEditing  === true 1`] = `
<div
  className="addPostForm"
>
  <div>
    <div
      className="postTitle"
    >
      Edit
       New Post
    </div>
    <form>
      <div>
        <label>
          ID:
        </label>
        <input
          disabled={true}
          type="text"
          value={2}
        />
        <br />
        <label>
          Title:
        </label>
         
        <input
          placeholder="Enter Title"
          type="text"
          value="test post title"
        />
         
        <br />
        <label>
          Description:
        </label>
        <textarea
          placeholder="Description"
          rows="5"
          value="test post body"
        />
      </div>
      <br />
      <button
        onClick={[MockFunction]}
      >
        Cancel
      </button>
      <button
        onClick={[MockFunction]}
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
